sources/Tests/tests/ECS/ecs.cpp

```diff
@@ -228,3 +229,135 @@ TEST_CASE("game_events_handling", "[ecs]")
   gameWorld->emitEvent(TestEvent{20});
   REQUIRE(listener->getLastMessageCode() == 10);
 }
+
+TEST_CASE("game_objects_transform_components_management", "[ecs]")
+{
+    std::shared_ptr<GameWorld> gameWorld = GameWorld::createInstance();
+
+    std::vector<GameObject> gameObjects;
+    for(int i = 0; i < 100; i++) {
+        GameObject object = gameWorld->createGameObject();
+        object.addComponent<TransformComponent>();
+        gameObjects.push_back(object);
+    }
+
+    for(int i = 50; i < 60; i++) {
+        gameWorld->removeGameObject(gameObjects[i]);
+    }
+
+    for(int i = 0; i < 100; i++) {
+        GameObject object = gameWorld->createGameObject();
+        object.addComponent<TransformComponent>();
+        gameObjects.push_back(object);
+    }
+
+    int iterations = 190;
+    gameWorld->forEach([&iterations](const GameObject& object) {
+        REQUIRE(object.isAlive());
+        iterations--;
+    });
+    REQUIRE(!iterations);
+
+    iterations = 190;
+    GameObject foundObj = gameWorld->findGameObject([&iterations](const GameObject& object) {
+        REQUIRE(object.isAlive());
+        iterations--;
+        return false;
+    });
+    REQUIRE(!iterations);
```

Comment:
According to the engine code style, it is recommended to avoid implicit integers to bools conversions. It will be better to use explicit comparison like `iterations == 0`.